### PR TITLE
Refactor List-All-Promotions API

### PR DESCRIPTION
**Pull request by iamnwi**
- Refactor `Lisll All Promotions` API
- Put get Home page API before Swagger Docs.
  - Otherwise, it doesn't work and gives you 404 Not Found
  - I guess its because the Flask RestPlus take over the control of parsing the route '/'

### CODE REVIEW

File: service/service.py [hunk 1]
```diff
@@ -28,37 +28,69 @@
...
-        promotions = Promotion.find_by_code(code)
-    else:
-        app.logger.info('Request for promotion list')
-        promotions = Promotion.all()
+api = Api(app,
+          version='1.0.0',
+          title='Promotion Demo REST API Service',
+          description='This is a sample Promotion server.',
+          default='promotions',
+          default_label='Pet shop operations',
```
Comment (reneesun) [inline service/service.py]: Maybe `Promotion operations`?

Comment (iamnwi) [reply service/service.py]: Oh, yes! Should be Promotion operations. Thanks for reviewing it closely.

File: service/service.py [hunk 2]
```diff
@@ -28,37 +28,69 @@
...
-        app.logger.info('Request for promotion list')
-        promotions = Promotion.all()
+api = Api(app,
+          version='1.0.0',
+          title='Promotion Demo REST API Service',
+          description='This is a sample Promotion server.',
+          default='promotions',
+          default_label='Promotion operations',
+          doc='/apidocs/',
+         )
```
[the remark this answers is not in the document]
Comment (iamnwi) [reply service/service.py]: Sure, or I can add this in the next issue. I just wonder if we really need that.

[the remark this answers is not in the document]
Comment (reneesun) [reply service/service.py]: I see some teams add the authorizations. But the requirement actually does not mention the security. What do you think?

[the remark this answers is not in the document]
Comment (iamnwi) [reply service/service.py]: Better to add this, but just wondering if we also need to add the authorizations header in our UI after adding this in our code. I will try adding this.

**Comment by reneesun** [comment]
IBM Cloud [toolchain](https://cloud.ibm.com/devops/toolchains/a903ceab-af58-4d40-9de5-39501ad15e1a?env_id=ibm%3Ayp%3Aus-south "nyu-devops-fall19-promotion-toolchain"): [Delivery Pipeline](https://cloud.ibm.com/devops/pipelines/5860bddd-fc68-4c80-afbc-55ea22efe279 "nyu-devops-fall19-promotion-pipeline") deployed [promotions](https://github.com/devops-fall-2019-promotions-squad/promotions/releases/tag/deploy-ibm-yp-us-south-rs6565-nyu.edu-dev-20191209-003910 "deploy-ibm-yp-us-south-rs6565-nyu.edu-dev-20191209-003910") to [dev](https://console.bluemix.net/apps/efcd7d31-ed8b-433d-b811-007435b814df?env_id=ibm:yp:us-south "ibm:yp:us-south:rs6565@nyu.edu:dev"), including commit e5479792edb564bcadc52e322e02dca55ccce04b

**Comment by reneesun** [comment]
IBM Cloud [toolchain](https://cloud.ibm.com/devops/toolchains/a903ceab-af58-4d40-9de5-39501ad15e1a?env_id=ibm%3Ayp%3Aus-south "nyu-devops-fall19-promotion-toolchain"): [Delivery Pipeline](https://cloud.ibm.com/devops/pipelines/5860bddd-fc68-4c80-afbc-55ea22efe279 "nyu-devops-fall19-promotion-pipeline") deployed [promotions](https://github.com/devops-fall-2019-promotions-squad/promotions/releases/tag/deploy-ibm-yp-us-south-rs6565-nyu.edu-prod-20191209-004350 "deploy-ibm-yp-us-south-rs6565-nyu.edu-prod-20191209-004350") to [prod](https://console.bluemix.net/apps/ce7c172b-4a93-4d4a-be2f-bbf7bd1fb4b7?env_id=ibm:yp:us-south "ibm:yp:us-south:rs6565@nyu.edu:prod"), including commit e5479792edb564bcadc52e322e02dca55ccce04b

**Pull request merged**